app/formats/json/PanoramaHistoryFormats.scala

```diff
@@ -0,0 +1,21 @@
+package formats.json
+
+import play.api.libs.json.{JsPath, Reads}
+import play.api.libs.functional.syntax._
+
+object PanoramaHistoryFormats {
+  case class IndividualPanoHistory(pano: String, year: Int, month: Int)
+  case class PanoHistorySubmission(currentId: String, history: Seq[IndividualPanoHistory], visitedTimestamp: String)
+
+  implicit val IndividualPanoHistoryReads: Reads[IndividualPanoHistory] = (
+    (JsPath \ "pano").read[String] and
+      (JsPath \ "year").read[Int] and
+      (JsPath \ "month").read[Int]
+    )(IndividualPanoHistory.apply _)
+
+  implicit val PanoHistorySubmissionReads: Reads[PanoHistorySubmission] = (
+    (JsPath \ "currentId").read[String] and
+      (JsPath \ "history").read[Seq[IndividualPanoHistory]] and
+      (JsPath \ "visitedTimestamp").read[String]
```

Comment:
You should look at how we handle other timestamps on the Explore or Validate pages to get an idea of how to read in the timestamps in an easier way. I believe that we read them as a `Long` and they get converted to a Timestamp rather easily in Scala; more easily than I saw you doing in `PanoramaHistoryTable.scala`